### PR TITLE
fix(itest): stabilize flaky beforeAll setups (wallet-service UTXO lag + blueprint timestamp collision)

### PR DESCRIPTION
**Pull request by pedroferreira1**
### Motivation

The integration suite fails intermittently in CI with errors that are **not product bugs** — they're timing races in one-time `beforeAll` setup code. Because they happen in a hook (not an `it()` body), `jest.retryTimes(2)` cannot recover them, so a single flaky setup send fails an entire test file. Both issues are pre-existing on `master` and affect every PR's CI equally; there are **no `src/` changes** in this PR (integration-test infrastructure only).

### Root cause 1 — wallet-service UTXO-index lag (the CI blocker)

`service-specific/utxos.test.ts` → `[Service] getUtxos` `beforeAll` issues two consecutive service sends whose input selection depends on the previous send's change output. `waitForTx` (→ `pollForTx`) only confirms a tx is **indexed** by the wallet-service; the service updates its **UTXO index** in a separate, non-atomic step. Sending the next tx before that catch-up makes input selection either:

- pick an already-spent UTXO → `Error sending tx proposal` (`sendTransactionWalletService.ts:962`) — seen in CI, or
- find nothing yet → `No UTXOs available for the token …` (`:459`).

Both are the same lag.

**Fix:** `pollForUtxoConsistency(wallet, tx)` — polls `getUtxos()` until the tx's spent inputs are gone **and** its own output is available — exposed as `adapter.waitForUtxoConsistency(...)` and used between the setup's consecutive sends.

## Root cause 2 — createOCBs blueprint timestamp collision

`createOCBs` (one-time `beforeAll` in `setupTests-integration.js`) deploys ~8 on-chain-blueprint txs back-to-back, spaced only by `waitTxConfirmed` (block confirmation, which the dev miner delivers sub-second). Occasionally two land in the same wall-clock second; the child selects the parent and the fullnode rejects it: `full validation failed: tx.timestamp == parent.timestamp`.

**Fix:** `waitUntilNextTimestamp` after each blueprint tx so every subsequent one lands in a strictly later second.

### Acceptance Criteria
-  Add `pollForUtxoConsistency` + `adapter.waitForUtxoConsistency`, and use it between the consecutive service sends in the `[Service] getUtxos` setup so it no longer selects a stale or not-yet-available UTXO.
- Space `createOCBs` blueprint deploys with `waitUntilNextTimestamp` so consecutive blueprint txs cannot share a wall-clock second.

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

## Summary by CodeRabbit

- **Tests**
  - Improved integration test synchronization by waiting for wallet-service UTXO indexing consistency after each send.
  - Added polling logic to ensure spent inputs disappear from the UTXO set and new transaction outputs become available before continuing.
  - Coordinated timestamps between on-chain blueprint transactions to enhance reliability.

- **Bug Fixes**
  - Reduced intermittent integration-test failures caused by stale UTXO data and edge cases involving transactions with identical timestamps.

### CODE REVIEW

File: __tests__/integration/helpers/service-facade.helper.ts [hunk 1]
```diff
@@ -152,6 +152,42 @@ export async function pollForTx(walletForPolling: HathorWalletServiceWallet, txI
...
+  wallet: HathorWalletServiceWallet,
+  tx: { hash?: string | null; inputs: { hash: string; index: number }[] }
+): Promise<void> {
+  const spent = new Set(tx.inputs.map(input => `${input.hash}:${input.index}`));
+  await pollUntilCondition(
+    async () => {
+      const { utxos } = await wallet.getUtxos();
+      const noStaleInput = utxos.every(utxo => !spent.has(`${utxo.tx_id}:${utxo.index}`));
+      const ownOutputAvailable = tx.hash != null && utxos.some(utxo => utxo.tx_id === tx.hash);
+      return noStaleInput && ownOutputAvailable;
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply __tests__/integration/helpers/service-facade.helper.ts:192]: done https://github.com/HathorNetwork/hathor-wallet-lib/pull/1129/changes/aca6aeb4194bca865558cd8cae2716a12cba8eae

**Review by tuliomir — APPROVED** [review]

**Review by raul-oliveira — APPROVED** [review]
